### PR TITLE
Implementation of PS-3969 - 5.7

### PR DESCRIPTION
**Pull request by altmannmarcelo**
This commit adds a variable to track when an audit log entry was either
 dropped or written directly to the file due to its size being bigger
than audit_log_buffer_size

### CODE REVIEW

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]
Same comments as for 5.6

**Comment by altmannmarcelo** [comment]
Ready for the next review - https://ps.cd.percona.com/job/percona-server-5.7-param/114/

File: plugin/audit_log/audit_log.c [hunk 1]
```diff
@@ -2063,7 +2065,10 @@ static struct st_mysql_audit audit_log_descriptor=
 
 static struct st_mysql_show_var audit_log_status_variables[]=
 {
-  {NullS, NullS, SHOW_LONG, SHOW_SCOPE_GLOBAL}
+  {"Audit_log_buffer_size_overflow",
+   (char*) &audit_log_buffer_size_overflow,
+   SHOW_INT, SHOW_SCOPE_GLOBAL},
```
Comment (laurynas-biveinis) [inline plugin/audit_log/audit_log.c]: It was SHOW_LONG in 5.6 version

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]
Same comments as for 5.6 + 1 5.7-specific comment

**Comment by altmannmarcelo** [comment]
Ready for next review - https://ps.cd.percona.com/job/percona-server-5.7-param/121/ (some Jenkins build failed, seems to be something with the pipeline itself)

File: plugin/audit_log/buffer.c [hunk 2]
```diff
@@ -190,6 +191,7 @@ int audit_log_buffer_write(audit_log_buffer_t *log, const char *buf, size_t len)
       log->write_func(log->write_func_data, buf, len, LOG_RECORD_COMPLETE);
       audit_log_buffer_resume(log);
     }
+    my_atomic_add64(&audit_log_buffer_size_overflow,1);
```
Comment (laurynas-biveinis) [inline plugin/audit_log/buffer.c]: space after comma:
```suggestion
    my_atomic_add64(&audit_log_buffer_size_overflow, 1);
```

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

**Comment by altmannmarcelo** [comment]
Ready for next review

**Review by laurynas-biveinis — APPROVED** [review]

**Comment by altmannmarcelo** [comment]
replaced by #3057

**Pull request closed without merging**